src/main/java/com/box/sdk/BoxWebHook.java

```diff
@@ -0,0 +1,686 @@
+        BoxAPIConnection api = target.getAPI();
+
+        BoxWebHook.TargetType type = type(target);
+        validateTriggers(type, triggers);
+
+        JsonObject targetJSON = new JsonObject()
+                .add(JSON_KEY_TARGET_TYPE, type.getValue())
+                .add(JSON_KEY_TARGET_ID, target.getID());
+
+        JsonObject requestJSON = new JsonObject()
+                .add(JSON_KEY_TARGET, targetJSON)
+                .add(JSON_KEY_ADDRESS, address.toExternalForm())
+                .add(JSON_KEY_TRIGGERS, toJsonArray(CollectionUtils.map(triggers, TRIGGER_TO_VALUE)));
+
+        URL url = WEBHOOKS_URL_TEMPLATE.build(api.getBaseURL());
+        BoxJSONRequest request = new BoxJSONRequest(api, url, HttpMethod.POST);
+        request.setBody(requestJSON.toString());
+
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+        JsonObject responseJSON = JsonObject.readFrom(response.getJSON());
+
+        BoxWebHook webHook = new BoxWebHook(api, responseJSON.get(JSON_KEY_ID).asString());
+        return webHook.new Info(responseJSON);
+    }
+
+    /**
+     * Helper function to create JsonArray from collection.
+     *
+     * @param values collection of values to convert to JsonArray
+     * @return JsonArray with collection values
+     */
+    private static JsonArray toJsonArray(Collection<String> values) {
+        JsonArray array = new JsonArray();
+        for (String value : values) {
+            array.add(value);
+        }
+        return array;
+
+    }
+
+    /**
+     * Returns iterator over all {@link BoxWebHook}-s.
+     *
+     * @param api
+     *            the API connection to be used by the resource
+     * @return existing {@link BoxWebHook.Info}-s
+     */
+    public static Iterable<BoxWebHook.Info> all(final BoxAPIConnection api) {
+        return new BoxResourceIterable<BoxWebHook.Info>(api, WEBHOOKS_URL_TEMPLATE.build(api.getBaseURL()), 64) {
+
+            @Override
+            protected BoxWebHook.Info factory(JsonObject jsonObject) {
+                BoxWebHook webHook = new BoxWebHook(api, jsonObject.get("id").asString());
+                return webHook.new Info(jsonObject);
+            }
+
+        };
+    }
+
+    /**
+     * Resolves {@link BoxWebHook.TargetType} for a provided {@link BoxResource}.
+     *
+     * @param target
+     *            for resolving
+     * @return resolved type
+     * @throws IllegalArgumentException
+     *             in case of unsupported {@link BoxResource}: only {@link BoxFolder} or {@link BoxFile} is supported.
+     */
+    private static TargetType type(BoxResource target) {
+        if (target instanceof BoxFolder) {
+            return TargetType.FOLDER;
+        } else if (target instanceof BoxFile) {
+            return TargetType.FILE;
+        } else {
+            throw new IllegalArgumentException(String.format("Unsupported BoxItem, expected: '%s' but it was '%s': ",
```

Comment:
You might want to use "Unsupported target resource..." instead of "Unsupported BoxItem" as the exception message.